### ISSUE TITLE
socket.io 3

### ISSUE THREAD

**Issue by nuclearace** [socketio/socket.io-client-swift#1308]
Need to support the changes in engine and socket.

**Comment by nuclearace** [comment]
Current timeline for getting this merged is either sometime this week, or next weekend, depending on how testing and other things go.

**Comment by davenquinn** [comment]
Is it possible to state prominently in the docs or README that Socket.IO v3 is not supported yet, until the support is merged and integrated into the published version? (and then state prominently the version at which Socket.IO v3 support starts)

Version 3 is now the default for Socket.IO proper, and the docs there do not do a great job of highlighting backwards incompatibilities, so I used it without thought for the server. Then, the way logging is handled by this library did not lead me in the right direction to diagnose the underlying issues. I understand this is an "unfortunate timing" sort of problem, and mostly upstream, but it was quite a nasty time trap for me (someone with little understanding of the transport protocols involved).

In the end, downgrading my server to version `2.3.0` fixed my issues with no other code changes on server or client.

**Comment by mohamnexus** [comment]
we wait for it..

**Comment by mohamnexus** [comment]
waiting... we need it

**Comment by wangyuanfen** [comment]
waiting... we need it

**Comment by RomanGalochkin** [comment]
When the third version will be supported?

**Comment by mouadroaui** [comment]
When the third version will be supported?

**Comment by nuclearace** [comment]
`v16.0.0` is been released.

**Comment by justlaputa** [comment]
hi, I know this is closed and 16.0.0 supports V3

It cost me a lot of time to debug my server code, but finally searched in client side source code, to find that v15.2 does not support V3 server.

there is no clear mention in README, this may cause some confusion, as it is conflicting with the [V3 doc](https://socket.io/blog/socket-io-3-1-0/):

> In order to ease the migration to Socket.IO v3, the v3 server is now able to communicate with v2 clients:
```
const io = require("socket.io")({
  allowEIO3: true // false by default
});
```

can we add some explicit mention in v15.x's README? so other people won't be confused. Thanks!
